lib/Github/HttpClient/Plugin/Authentication.php

```diff
@@ -33,18 +33,30 @@ public function __construct($tokenOrLogin, $password, $method)
     public function doHandleRequest(RequestInterface $request, callable $next, callable $first)
     {
         switch ($this->method) {
+            case Client::AUTH_CLIENT_ID:
             case Client::AUTH_HTTP_PASSWORD:
+                if ($this->method === Client::AUTH_HTTP_PASSWORD) {
+                    @trigger_error('Using the "Client::AUTH_HTTP_PASSWORD" authentication method is deprecated in knp-labs/php-github-api v2.15 and will be removed in knp-labs/php-github-api v3.0. Use "Client::AUTH_ACCESS_TOKEN" instead.', E_USER_DEPRECATED);
+                }
+
                 $request = $request->withHeader(
                     'Authorization',
                     sprintf('Basic %s', base64_encode($this->tokenOrLogin.':'.$this->password))
                 );
                 break;
 
+            case Client::AUTH_ACCESS_TOKEN:
             case Client::AUTH_HTTP_TOKEN:
+                if ($this->method === Client::AUTH_HTTP_TOKEN) {
```

Comment:
You can avoid this guard by changing the case order.

```php
case Client::AUTH_HTTP_TOKEN:
@trigger_error('Using the "Client::AUTH_HTTP_TOKEN" authentication method is deprecated in knp-labs/php-github-api v2.15 and will be removed in knp-labs/php-github-api v3.0. Use "Client::AUTH_ACCESS_TOKEN" instead.', E_USER_DEPRECATED);
case Client::AUTH_ACCESS_TOKEN:
$request = $request->withHeader('Authorization', sprintf('token %s', $this->tokenOrLogin));
break;
```

Reply:
Hmm, indeed! That's actually a good idea!